lib/Graph/NodeValue.cpp

```diff
@@ -51,6 +53,11 @@ void NodeValue::replaceAllUsesOfWith(NodeValue v, const Function *F) const {
     assert(site->getResNo() == getResNo() && "Invalid list of uses");
     site->setOperand(v.getNode(), v.getResNo());
   }
+
+  // Log all nodes replacement.
+  if (getNode()->getParent()) {
+    LOG_NODE_REPLACE(getNode(), v.getNode())
```

Comment:
Is this the right level to log at? replaceAllUsesOfWith replaces one NodeValue with another and that information would be lost here.

Reply:
Yes, we record the nodes replacement by the last node and record all newly created nodes before replaceAllUsesOfWith() by logging at where we create these nodes (F->addNode() now)